packages/inspector/src/handler.ts

```diff
@@ -60,14 +60,32 @@ export class InspectionHandler implements IDisposable, IInspector.IInspectable {
     }
 
     if (this._editor && !this._editor.isDisposed) {
-      this._editor.model.value.changed.disconnect(this.onTextChanged, this);
+      this._selectionsMonitor.activityStopped.disconnect(
+        this.onEditorChange,
+        this
+      );
+      this._valueMonitor.activityStopped.disconnect(this.onEditorChange, this);
     }
     let editor = (this._editor = newValue);
     if (editor) {
       // Clear the inspector in preparation for a new editor.
       this._cleared.emit(void 0);
-      editor.model.value.changed.connect(
-        this.onTextChanged,
+
+      this._selectionsMonitor = new ActivityMonitor({
+        signal: editor.model.selections.changed,
+        timeout: 1000
+      });
+      this._selectionsMonitor.activityStopped.connect(
+        this.onEditorChange,
+        this
+      );
+
+      this._valueMonitor = new ActivityMonitor({
+        signal: editor.model.value.changed,
+        timeout: 1000
+      });
+      this._valueMonitor.activityStopped.connect(
+        this.onEditorChange,
         this
       );
     }
```

Comment:
It is possible to select a new cell/editor without changing the value or cursor position, in which case the inspector won't update. I think we should call `onEditorChange()` here as well to catch that case.

Reply:
Gotchu. Would that not trigger a `editor.model.selections.changed`? Is there another event that I should use?

Reply:
If the selection of the new cell doesn't move the cursor, it won't trigger a change signal (I was able to verify by clicking on the same cursor position in a cell repeatedly). But you needn't listen to another event for that, we can just to call the `onEditorChange()` function in this setter directly.